gst/nnstreamer/nnstreamer_conf.c

```diff
@@ -174,22 +174,34 @@ _get_filenames (nnsconf_type_path type, const gchar * dir, GSList ** listF,
     GSList ** listB, guint * counter)
 {
   GDir *gdir;
-  const gchar *name;
+  const gchar *entry;
   gchar *fullpath;
+  gchar *basename;
+  gchar *name;
+  gsize prefix, extension, len;
 
   if ((gdir = g_dir_open (dir, 0U, NULL)) == NULL)
     return FALSE;
 
-  while (NULL != (name = g_dir_read_name (gdir))) {
+  prefix = strlen (subplugin_prefixes[type]);
+  extension = strlen (NNSTREAMER_SO_FILE_EXTENSION);
+
+  while (NULL != (entry = g_dir_read_name (gdir))) {
     /* check file prefix for given type, currently handle .so and .dylib. */
-    if (g_str_has_prefix (name, subplugin_prefixes[type]) &&
-        g_str_has_suffix (name, NNSTREAMER_SO_FILE_EXTENSION)) {
-      fullpath = g_build_filename (dir, name, NULL);
+    if (g_str_has_prefix (entry, subplugin_prefixes[type]) &&
+        g_str_has_suffix (entry, NNSTREAMER_SO_FILE_EXTENSION)) {
+      fullpath = g_build_filename (dir, entry, NULL);
 
       if (_validate_file (type, fullpath)) {
+        basename = g_path_get_basename (entry);
+        len = strlen (basename) - prefix - extension;
+        name = g_strndup (basename + prefix, len);
+
         *listF = g_slist_prepend (*listF, fullpath);
-        *listB = g_slist_prepend (*listB, g_path_get_basename (name));
+        *listB = g_slist_prepend (*listB, name);
```

Comment:
How about change list name to match the changes?

Reply:
I updated description about subplugin name.